packages/api/pages/api/languages/[code]/import.ts

```diff
@@ -29,23 +61,45 @@ export default createRoute()
         return;
       }
 
-      // trigger import process
-      const response = await fetch(importTriggerUrl);
-      const result = await response.json();
-      // TODO: actually use the result
-      console.log(result);
+      const currentJob = await client.languageImportJob.findUnique({
+        where: {
+          languageId: language.id,
+        },
+      });
 
-      res.ok();
-    },
-  })
-  .get<void, void>({
-    authorize: authorize((req) => ({
-      action: 'administer',
-      subject: 'Language',
-      subjectId: req.query.code,
-    })),
-    async handler(req, res) {
-      // TODO: query DB for import status
+      if (currentJob && !currentJob.endDate) {
+        return res.conflict([
+          {
+            code: 'JobAlreadyRunning',
+          },
+        ]);
+      }
+
+      await client.languageImportJob.upsert({
```

Comment:
The import request creates a new job if one isn't already running

Reply:
Just making sure I'm following: it's an upsert instead of insert because there could be a previously finished job for this language?